<h1>Palmer Peguin Data Analytics</h1>
<hr>
<h3>Analyst: Agon, Lance D.</h3>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Import dataset to Jupyter notebook</p>
<p>converting dataset to dataframe</p>
<ul><strong>Requirements</strong>
    <li>Name of the Dataframe</li>
    <li>Source</li>
</ul>


In [2]:
pengu = pd.read_csv('DATASETS\\penguins.csv')

In [3]:
pengu

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h2>Adding Column.</h2>

In [4]:
pengu['mass_kg'] = np.round(pengu['body_mass_g'] / 1000,1)

In [5]:
pengu

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
0         3750.0    male  2007      3.8  
1         3800.0  female  2007      3.8  
2         3250.0  female  2007      3.2  
3            NaN     NaN  2007      NaN  
4         3450.0  female  2007      3.4  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[344 rows x 9 columns]

In [6]:
pengu.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year     mass_kg  
count   344.000000  342.000000  
mean   2008.029070    4.202047  
std       0.818356    0.803089  
min    2007.000000    2.700000  
25%    2007.000000    3.600000  
50%    2008.000000    4.000000  
75%    2009.000000    4.800000  
max    2009.000000    6.300000

In [7]:
pengu.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   
mass_kg            342.0     4.202047    0.803089     2.7     3.600     4.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0  
mass_kg               4.8     6.3

<H2>Cleaning the Data</H2>

In [8]:
pengu.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
mass_kg               2
dtype: int64

<h4>Selecting a column<h4>
<h4>Syntax</h4><br>
data_frame['column_name']

<h5>Identify the element values of a column.</h5>

In [9]:
pengu['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

<H5>Creating Dataframe for Specific Species.</H5>

In [10]:
gentoo_df = pengu[pengu['species'] == 'Gentoo']

In [11]:
adelie_df = pengu[pengu['species'] == 'Adelie']

In [12]:
chinstrap_df = pengu[pengu['species'] == 'Chinstrap']

In [13]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  mass_kg  
152       4500.0  female  2007      4.5  
153       5700.0    male  2007      5.7  
154       4450.0  female  2007      4.4  
155       5700.0    male  2007      5.7  
156       5400.0    male  2007      5.4  
..           ...     ...   ...      ...  
271          NaN     NaN  2009      NaN  
272       4850.0  female  2009      4.8  
273       5750.0    male  2009      5.8  
274       5200.0  female  2009      5.2  
275       5400.0    male  2009      5.4  

[124 rows x 9 columns]

In [14]:
ch_heavy = chinstrap_df[chinstrap_df['body_mass_g'] >= 4000]

In [15]:
ch_heavy

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  mass_kg  
284       4150.0  female  2007      4.2  
289       4050.0    male  2007      4.0  
291       4050.0    male  2007      4.0  
295       4400.0    male  2007      4.4  
301       4150.0    male  2007      4.2  
305       4550.0    male  2008      4.6  
307       4300.0    male  2008      4.3  
309       4100.0    male  2008      4.1  
313       4800.0    male  2008      4.8  
315       4500.0    male  2008      4.5  
321       4450.0    male  2009      4.4  
323       4300.0    male  2009      4.3  
329       4050.0    male  2009      4.0  
333       4050.0    male  2009      4.0  
339       4000.0    male  2009      4.0  
342       4100.0    male  2009      4.1

In [16]:
ch_heavy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
 8   mass_kg            16 non-null     float64
dtypes: float64(5), int64(1), object(3)
memory usage: 1.2+ KB


<h2>Filling up the Null Values</h2>

In [17]:
pengu.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
mass_kg               2
dtype: int64

<li>bill length</li>

In [18]:
bill_length_ave = pengu['bill_length_mm'].mean()

In [19]:
pengu['bill_length_mm']= pengu['bill_length_mm'].fillna(bill_length_ave)

<li>bill depth</li>

In [20]:
bill_depth_ave = pengu['bill_depth_mm'].mean()

In [21]:
pengu['bill_depth_mm']= pengu['bill_depth_mm'].fillna(bill_depth_ave)

<li>flipper length</li>

In [22]:
flipper_length_ave = pengu['flipper_length_mm'].mean()

In [23]:
pengu['flipper_length_mm']= pengu['flipper_length_mm'].fillna(bill_depth_ave)

<li>body mass</li>

In [24]:
body_mass_ave = pengu['body_mass_g'].mean()

In [25]:
pengu['body_mass_g']= pengu['body_mass_g'].fillna(body_mass_ave)

<li>mass kg</li>

In [26]:
mass_kg_ave = pengu['mass_kg'].mean()

In [27]:
pengu['mass_kg']= pengu['mass_kg'].fillna(mass_kg_ave)

<li>sex</li>

In [28]:
pengu['sex']=pengu['sex'].fillna(pengu['sex'].mode()[0])

In [29]:
pengu.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
mass_kg              0
dtype: int64

<h2>Changing the Datatypes.</h2>

In [30]:
pengu['species'] = pengu['species'].astype('category')
pengu['island'] = pengu['island'].astype('category')
pengu['year'] = pengu['year'].astype('category')
pengu['sex'] = pengu['sex'].astype('category')

<h1>Exploring Data Analysist</h1>

<h2>All about population.</h2>

<h3>1. What species has the highest population?</h3>

In [31]:
species_counts = pengu['species'].value_counts()
species_counts

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h4>Insight #1</h4>
<h4>The Highest species population is Adelie which has 152.</h4>
<hr>

<h3>2. What percentage of the total population does each species represent?</h3>

In [32]:
species_percentage = np.round((species_counts / species_counts.sum()) * 100,1)
species_percentage

species
Adelie       44.2
Gentoo       36.0
Chinstrap    19.8
Name: count, dtype: float64

<h4>Insight #2</h4>
<h4>The population is unevenly distributed among the species, with Adelie making up almost 50% of the total population.</h4>
<hr>

<h3>3.What is the most common species in the Dream Island?</h3>

In [33]:
species_dream = pengu[pengu['island'] == 'Dream']

In [34]:
species_dream['species'].value_counts()

species
Chinstrap    68
Adelie       56
Gentoo        0
Name: count, dtype: int64

<h4>Insight #3</h4>
<h4>There are two species in this island Chistrap and Adelie. The most common species in the Dream Island is Chistrap.</h4>
<hr>

<h3>4.What is the most common species in the Biscoe Island?</h3>

In [35]:
species_biscoe = pengu[pengu['island'] == 'Biscoe']

In [36]:
species_biscoe['species'].value_counts()

species
Gentoo       124
Adelie        44
Chinstrap      0
Name: count, dtype: int64

<h4>Insight #4</h4>
<h4>There are two species in this island Adelie and Gentoo. The most common species in the Biscoe Island is Gentoo.</h4>
<hr>

<h3>5.What is the most common species in the Torgersen Island?</h3>

In [37]:
species_torgersen = pengu[pengu['island'] == 'Torgersen']

In [38]:
species_torgersen['species'].value_counts()

species
Adelie       52
Chinstrap     0
Gentoo        0
Name: count, dtype: int64

<h4>Insight #5</h4>
<h4>The only species in this island is Adelie and is has 52 population.</h4>
<hr>

<h3>6.What is the gender breakdown of the population of each species?</h3>

In [39]:
chins_sex_bd = chinstrap_df['sex'].value_counts()
adelie_sex_bd = adelie_df['sex'].value_counts()
gentoo_sex_bd = gentoo_df['sex'].value_counts()

In [40]:
print("Adelie:\n", adelie_sex_bd)
print("\nChinstrap:\n", chins_sex_bd)
print("\nGentoo:\n", gentoo_sex_bd)

Adelie:
 sex
male      73
female    73
Name: count, dtype: int64

Chinstrap:
 sex
female    34
male      34
Name: count, dtype: int64

Gentoo:
 sex
male      61
female    58
Name: count, dtype: int64


<h4>Insight #6</h4>
<h4>The Adelie and Chinstrap penguin populations have an equal number of males and females, while Gentoo penguins have slightly more males than females, showing that overall, the gender distribution is fairly balanced across all species.</h4>
<hr>

<h3>7.Based on the population size of each species, which species has the highest average bill length?</h3>

In [41]:
chin_ave_bill_length = round(chinstrap_df['bill_length_mm'].mean(),1)
adel_ave_bill_length = round(adelie_df['bill_length_mm'].mean(),1)
gent_ave_bill_length = round(gentoo_df['bill_length_mm'].mean(),1)

In [42]:
print("Adelie:\n", adel_ave_bill_length)
print("\nChinstrap:\n", chin_ave_bill_length)
print("\nGentoo:\n", gent_ave_bill_length)

Adelie:
 38.8

Chinstrap:
 48.8

Gentoo:
 47.5


<h4>Insight #7</h4>
<h4>Chinstrap penguins have the highest average bill length at 48.8 mm, followed by Gentoo penguins with 47.5 mm, while Adelie penguins have the shortest average bill length at 38.8 mm.</h4>
<hr>

<h1>Aggregating Data</h1>

<h2>Using .groupby()</h2>

<h3>1. How many species in every island?</h3>

In [43]:
pengu.groupby(['species','sex','island']).size()

species    sex     island   
Adelie     female  Biscoe       22
                   Dream        27
                   Torgersen    24
           male    Biscoe       22
                   Dream        29
                   Torgersen    28
Chinstrap  female  Biscoe        0
                   Dream        34
                   Torgersen     0
           male    Biscoe        0
                   Dream        34
                   Torgersen     0
Gentoo     female  Biscoe       58
                   Dream         0
                   Torgersen     0
           male    Biscoe       66
                   Dream         0
                   Torgersen     0
dtype: int64

In [44]:
avg_massspecies_island = round(pengu.groupby(['species','island','sex'])['mass_kg'].mean(),1)

In [45]:
avg_massspecies_island

species    island     sex   
Adelie     Biscoe     female    3.4
                      male      4.1
           Dream      female    3.3
                      male      4.0
           Torgersen  female    3.4
                      male      4.0
Chinstrap  Biscoe     female    NaN
                      male      NaN
           Dream      female    3.5
                      male      3.9
           Torgersen  female    NaN
                      male      NaN
Gentoo     Biscoe     female    4.7
                      male      5.4
           Dream      female    NaN
                      male      NaN
           Torgersen  female    NaN
                      male      NaN
Name: mass_kg, dtype: float64

<h1>Testing Correlation</h1>

<h2>Pearson R Correlation</h2>
<hr>
<h3>Used for testing correlation between two <strong>NUMERICAL VALUES</strong> </h3>

In [46]:
Blen_Bdept = pengu['bill_length_mm'].corr(pengu['bill_depth_mm'])

In [47]:
Blen_Bdept

np.float64(-0.23505287035553282)

<h4>Insight #1</h4>
<h4>There is no correlation between the bill length and bill depth.</h4>
<hr>

<h1>Correlation Test; Numerical - Categorial</h1>
<h2>Libary - SciPy (scientific python</h2>
<hr>
<h3>from scipy.stats import f_oneway</h3>

In [48]:
from scipy.stats import f_oneway

<h3>1. Is there a correlation between sex and body mass?</h3>

In [49]:
pengu[pengu['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [50]:
pengu[pengu['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [51]:
res, corr_sex_bodymass = f_oneway(pengu[pengu['sex'] == 'male']['body_mass_g'],pengu[pengu['sex'] == 'female']['body_mass_g'])

In [52]:
corr_sex_bodymass

np.float64(3.0226600487050246e-15)

<h4>Insight #1</h4>
<h4>Pvalue is equal to 3.02, this means there are correlation between sex and body mass of the penguins.</h4>
<hr>

<h3>2. Is there a correlation between species and body mass?</h3>

In [53]:
pengu[pengu['species'] == 'Adelie']['body_mass_g']

0      3750.000000
1      3800.000000
2      3250.000000
3      4201.754386
4      3450.000000
          ...     
147    3475.000000
148    3450.000000
149    3750.000000
150    3700.000000
151    4000.000000
Name: body_mass_g, Length: 152, dtype: float64

In [54]:
pengu[pengu['species'] == 'Chinstrap']['body_mass_g']

276    3500.0
277    3900.0
278    3650.0
279    3525.0
280    3725.0
        ...  
339    4000.0
340    3400.0
341    3775.0
342    4100.0
343    3775.0
Name: body_mass_g, Length: 68, dtype: float64

In [55]:
pengu[pengu['species'] == 'Gentoo']['body_mass_g']

152    4500.000000
153    5700.000000
154    4450.000000
155    5700.000000
156    5400.000000
          ...     
271    4201.754386
272    4850.000000
273    5750.000000
274    5200.000000
275    5400.000000
Name: body_mass_g, Length: 124, dtype: float64

In [56]:
res, corr_species_bodymass = f_oneway(pengu[pengu['species'] == 'Adelie']['body_mass_g'],pengu[pengu['species'] == 'Chinstrap']['body_mass_g'],pengu[pengu['species'] == 'Gentoo']['body_mass_g'])

In [57]:
corr_species_bodymass

np.float64(1.0069770504490732e-81)

<h4>Insight #2</h4>
<h4>Pvalue is equal to 1.00, this means there are correlation between species and body mass of the penguins.</h4>
<hr>

<h3>2. Is there a correlation between species and flipper length?</h3>